caom2-tap-server/src/main/java/ca/nrc/cadc/tap/caom2/ArtifactURI2URLFormat.java

```diff
@@ -141,23 +143,12 @@ public String format(Object object)
         log.debug("format: " + object + "," + jobID);
         if (object == null)
             return "";
-        StringBuilder sb = new StringBuilder();
-
         if (object instanceof String)
         {
-            try
-            {
-                URI uri = new URI((String) object);
-                if ("ad".equals(uri.getScheme()))
-                {
-                    sb.append(accessURL.toExternalForm());
-                    sb.append("/");
-                    sb.append(uri.getSchemeSpecificPart()); // ad URI
-                    sb.append("?RUNID=");
-                    sb.append(jobID);
-                }
-                else
-                    sb.append(uri.toASCIIString()); // pass-through (http, vos, etc)
+           try {
```

Comment:
You can't just add the RUNID to the URI - it would alweays be a bad idea because it isn't part of the identifier.

In this case, you need to call  on the CaomArtifactResolver.setRunID(...) and it will append it to the URL if the URL supports it (because some external URLs don't).

Otherwise - good.